plugins/appwrite/appwrite.go

```diff
@@ -0,0 +1,39 @@
+package appwrite
+
+import (
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/needsauth"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+)
+
+func AppwriteCLI() schema.Executable {
+	return schema.Executable{
+		Name:    "Appwrite CLI",
+		Runs:    []string{"appwrite"},
+		DocsURL: sdk.URL("https://appwrite.io/docs/command-line-ci"),
+		NeedsAuth: needsauth.IfAll(
+			needsauth.NotForHelpOrVersion(),
+			needsauth.NotWithoutArgs(),
+			needsauth.NotWhenContainsArgs("client"),
+			needsauth.NotWhenContainsArgs("login"),
+			needsauth.NotWhenContainsArgs("logout"),
+			needsauth.NotForExactArgs("deploy"),
+			needsauth.NotForExactArgs("projects"),
```

Comment:
Do I understand correctly that both `projects` and `projects <subcommand>` don't require authentication of any type? If so, we can switch to `NotWhenContainsArgs`

Reply:
No, `projects` command will give you the available subcommands without requiring a login. But anything you want to do with any of the projects, you need to be signed in. (same goes for almost all the commands above)

Reply:
Thanks for clarifying. In the current setup, `appwrite projects list` requires authentication. Could we switch to `NotWhenContainsArgs` so that we can not require authentication that command? And other similar subcommands? I say this because `projects list` wouldn't work with the project-specific API key that we are using for authentication.

Let me know if something about my question/concern seems unclear.

Reply:
>And other similar subcommands?

I think it's just projects that needs to be switched to `NotWhenContainsArgs`. `storage` and `databases` subcommands need auth.

Reply:
Thanks @arunsathiya. I've pushed the change.